### PR TITLE
deals with .xls older file extension

### PR DESCRIPTION
**Pull request by FredericBGA**
I need to deal with old .xls files for some users.
I'm not sure if all lines of code are really mandatory.

### CODE REVIEW

File: lib/galaxy/datatypes/binary.py [hunk 1]
```diff
@@ -1120,6 +1120,40 @@ def sniff( self, filename ):
 Binary.register_sniffable_binary_format("xlsx", "xlsx", Xlsx)
 
 
+class Xls( Binary ):
+    """Class describing an Excel (xls) file"""
+    file_ext = "xls"
+
+    def sniff( self, filename ):
+        ret = os.popen("file -i " + filename)
+        mimeType = ret.read().rstrip()
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/binary.py]: Please use the [subprocess module](https://docs.python.org/2/library/subprocess.html) instead, e.g. `mimeType = subprocess.check_output("file -i %s" % filename).rstrip()`

File: lib/galaxy/datatypes/binary.py [hunk 2]
```diff
@@ -1120,6 +1120,40 @@ def sniff( self, filename ):
...
+        if (mimeType.find("application/vnd.ms-excel") != -1):
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/binary.py]: The external parentheses are not needed.

**Review by nsoranzo — COMMENTED** [review]
You will also have to change the line:
```
<datatype extension="xls" type="galaxy.datatypes.tabular:Tabular"/>
```
in `config/datatypes_conf.xml.sample`
to:
```
<datatype extension="xls" type="galaxy.datatypes.binary:Xls" display_in_upload="true" />
```

File: lib/galaxy/datatypes/binary.py (hunk 1 above)
Comment (FredericBGA) [reply lib/galaxy/datatypes/binary.py]: ok, I'll change this

File: lib/galaxy/datatypes/binary.py (hunk 2 above)
Comment (FredericBGA) [reply lib/galaxy/datatypes/binary.py]: I'll change this too. 
I'm not yet really good with Python...

**Comment by nsoranzo** [comment]
@galaxybot test this

**Comment by nsoranzo** [comment]
Ping @blankenberg which initially added the `xls` fake datatype in commit c8a71a49f865f2141eaf8be341f42e5ba14f445d.

**Comment by jmchilton** [comment]
I have opened a PR with a variety of changes against the source branch here https://github.com/FredericBGA/galaxy/pull/1. @FredericBGA please have a look at the details and let me know if everything there looks okay. I'm not unwilling to drop support for legacy XLS datatype within Galaxy - I can't recall what it is used for - but the approach I PR'd would prevent us from needing to so if that is good enough we don't even need to worry about it further.

**Comment by nsoranzo** [comment]
And let's ping @blankenberg again!

**Comment by blankenberg** [comment]
IIRC the tabular `xls` type was added as part of the RGenetics work that I think has since been deprecated, although there may still be instances that are using these datatypes, and I'd predict that some of the tools will make a comeback at some point. 

I'm always in favor of maintaining backwards compatibility, so creating a new datatype seems the safer route here. But I don't have a specific argument for keeping the tabular based xls filetype around.

The PR by @jmchilton is required to have the `file` command work correctly on a mac. I'm not a huge fan of using `file` in general for these tests, I'd rather check that the magic bytes are a ms doc, then do the required deeper investigation into the file to determine the subtype. 

Regardless, the call to subprocess should be using an argument list, and not string formatting, since it will fail when there are e.g. spaces in the path.

**Pull request merged**